src/services/inlayHints.ts

```diff
@@ -226,14 +229,14 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
             return;
         }
 
-        for (let i = 0; i < args.length; ++i) {
-            const originalArg = args[i];
+        let pos = 0;
+        for (const originalArg of args) {
             const arg = skipParentheses(originalArg);
             if (shouldShowLiteralParameterNameHintsOnly(preferences) && !isHintableLiteral(arg)) {
                 continue;
             }
 
-            const identifierNameInfo = checker.getParameterIdentifierNameAtPosition(signature, i);
+            const identifierNameInfo = checker.getParameterIdentifierNameAtPosition(signature, pos++);

```

Comment:
I think you also don't want to increment unconditionally here, since we could be passing an empty tuple. Like so:

```ts
function foo1(a: unknown, b: unknown, c: unknown) { }
function bar1(...x: []) { // Empty tuple
    foo1(...x, 3, 4, 5); // Wrong: currently `foo1(/*a*/...x, /*b*/3, /*c*/4, 5)`, should be `foo1(...x, /*a*/3, /*b*/4, /*c*/5)`
}
```